src/tools/storage-key-recipe-resolver.ts

```diff
@@ -78,14 +78,14 @@ export class StorageKeyRecipeResolver {
   }
 
   /**
-   * Create stores with keys for all create handles.
+   * Create stores with keys for all create handles with ids.
    *
    * @param recipe should be long running.
    * @param arc Arc is associated with current recipe.
    */
   async createStoresForCreateHandles(recipe: Recipe, arc: Arc) {
     const resolver = new CapabilitiesResolver({arcId: arc.id});
-    for (const createHandle of recipe.handles.filter(h => h.fate === 'create')) {
+    for (const createHandle of recipe.handles.filter(h => h.fate === 'create' && !!h.id)) {
```

Comment:
Can we have a test with a create handle without id and ensuring storage key is not created?

Reply:
Sure thing.